services/builder/serviceCaller.go

```diff
@@ -75,12 +75,27 @@ func buildConfig(config *pb.Config, c pb.ContextBoxServiceClient, isTmpConfig bo
 
 // destroyConfig destroys existing clusters infra for a config using terraformer and kuber
 func destroyConfig(config *pb.Config, c pb.ContextBoxServiceClient) error {
+	err := destroy(config, c)
+	if err != nil {
+		return err
+	}
+	// save changes to DB
+	err = cbox.SaveConfigBuilder(c, &pb.SaveConfigRequest{Config: config})
+	if err != nil {
+		config.CurrentState = config.DesiredState
```

Comment:
Why are we doing this assignment ? `config.CurrentState = config.DesiredState`, The config its not being used after calling `destroyConfig` so this has no effect.

`destroy`, and `destroyConfig` are ambiguous names when first looking at them, maybe a more suiting name would be `destroyWithoutSave` and `destroyWithSave` as they essentially do the same thing but one saves to the DB and the other doesn't

Reply:
I agree with you but I was just following the pattern of updating `currentState` before calling the `SaveConfigBuilder` function. I think we should move the `currentState` updation to `SaveConfigBuilder` fucntion since fields like checksums and TTLs are being updated there.

Let me know your thoughts @MiroslavRepka

Reply:
I've actually addressed this line of code as part of fixing the document deletion #309